### PR TITLE
Rest-API

### PR DESCRIPTION
**Pull request by ezdac**
This implements a RESTful API as discussed in #355,  #357 and #10 
The encoding complies to the latest discussion in #357, while especially the endpoints need more
work and are incomplete.

It also implements a first [proposal](https://github.com/raiden-network/raiden/commit/b462cd466ce8b6a98024d0650f59f824a08a56b2) of the construction and forwarding of Events from raiden to the client.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -13,3 +13,8 @@ ethereum-serpent
 repoze.lru
 gevent-websocket==0.9.4
 cachetools>=2.0.0,<3.0.0
+sphinx
```
Comment (LefterisJP) [inline requirements.txt]: Sphinx is already given as a requirement in `requirements-dev.txt` and so is `bumpversion`.

We have the developer and user requirements separated, please remove them from here. This is also why the tests fail from the get-go.

Comment (ezdac) [inline requirements.txt]: That was quick! Did that already ;)

Comment (LefterisJP) [inline requirements.txt]: yep I noticed :)

**Comment by LefterisJP** [comment]
I am rebasing on top of the latest master changes. Lots of asset/token conflicts.

**Comment by LefterisJP** [comment]
Superseded by #388

**Pull request closed without merging**